rust/main/hyperlane-base/src/settings/checkpoint_syncer.rs

```diff
@@ -225,28 +233,12 @@ mod test {
 
         // Initialize a new checkpoint syncer and expect it to panic due to the reorg event.
         // `AssertUnwindSafe` is required for ignoring some type checks so the panic can be caught.
-        let startup_result = AssertUnwindSafe(checkpoint_syncer_conf.build_and_validate(None))
-            .catch_unwind()
-            .await
-            .unwrap_err();
-        if let Some(err) = startup_result.downcast_ref::<String>() {
-            assert_eq!(
-                *err,
-                r#"A reorg event has been detected: ReorgEvent {
-    local_merkle_root: 0x8da44bc8198e9874db215ec2000037c58e16918c94743d70c838ecb10e243c64,
-    canonical_merkle_root: 0xb437b888332ef12f7260c7f679aad3c96b91ab81c2dc7242f8b290f0b6bba92b,
-    checkpoint_index: 56,
-    unix_timestamp: 1620000000,
-    reorg_period: Blocks(
-        5,
-    ),
-}. Please resolve the reorg to continue."#
-            );
-        } else {
-            panic!(
-                "Caught panic has a different type than the expected one (`String`): {:?}",
-                startup_result
-            );
+        let result = checkpoint_syncer_conf.build_and_validate(None).await;
+        match result {
```

Comment:
not sure if there's a better way to test for equality tbh given I can't have `CheckpointSyncerBuildError` be partialeq with the eyre::Report, this feels ugly